### ISSUE TITLE
fine tune do not change model output

### ISSUE THREAD

**Issue by darcula1993** [GantMan/nsfw_model#33]
Hi Thank you very much for your pre-trained model!

I am gathering some new data and using your code to fine-tune.After 100 epochs, I get a new model (one weird thing is that the size of new model reduce a lot.....). However when I test it, it seems that new model behave exactly same as the pre-trained one(same acc,same recall and same precision for each class).

Any idea why it happens? I am checking your code and seems that you only set one layer trainable?

**Comment by GantMan** [comment]
Hi!  Which model did you train?  There are two here.  The mobilenet and the inception v3.  The mobilenet is significantly smaller in size.

Additionally, I'm setting a layer and all layers after that as trainable.   Take a look with model.summary()

this code
```
model.trainable = False
set_trainable = False
for layer in model.layers:
    if layer.name == 'conv2d_56':
        set_trainable = True
    if set_trainable:
        layer.trainable = True
    else:
        layer.trainable = False
```
flips a flag to make all follow-up layers trainable.

**Comment by darcula1993** [comment]
I am training InvceptionV3. I used the default code and parameter to train. The model summary looks fine but still got no model change.
I also notice that the new model is significant smaller than the pre-trained one, which is also weird.

The platform I am using is python 3.6, RTX 2080TI (XLA_GPU) and tf 2.0, cuda 10.0, cudnn 7.4.1

**Comment by darcula1993** [comment]
Since I have a large enough dataset (~40GB). I am trying to unfrozen all layer and train again. Let's see if it work or not.

**Comment by devinhee** [comment]
> Hi Thank you very much for your pre-trained model!
> 
> I am gathering some new data and using your code to fine-tune.After 100 epochs, I get a new model (one weird thing is that the size of new model reduce a lot.....). However when I test it, it seems that new model behave exactly same as the pre-trained one(same acc,same recall and same precision for each class).
> 
> Any idea why it happens? I am checking your code and seems that you only set one layer trainable?

I have same issue with your,  and I suggest you read Keras official documents.
I replace this code:
  `model.trainable = False
set_trainable = False
for layer in model.layers:
    if layer.name == 'conv2d_56':
        set_trainable = True
    if set_trainable:
        layer.trainable = True
    else:
        layer.trainable = False`
TO：
  `for layer in model.layers[:168]:
       model.trainable = False
   for layer in model.layers[168:]:
       model.trainable = True`
It works!
I don't think there is a problem with original codes, but I haven't figured out what's the problem yet.

**Comment by darcula1993** [comment]
> > Hi Thank you very much for your pre-trained model!
> > I am gathering some new data and using your code to fine-tune.After 100 epochs, I get a new model (one weird thing is that the size of new model reduce a lot.....). However when I test it, it seems that new model behave exactly same as the pre-trained one(same acc,same recall and same precision for each class).
> > Any idea why it happens? I am checking your code and seems that you only set one layer trainable?
> 
> I have same issue with your, and I suggest you read Keras official documents.
> I replace this code:
> `model.trainable = False set_trainable = False for layer in model.layers: if layer.name == 'conv2d_56': set_trainable = True if set_trainable: layer.trainable = True else: layer.trainable = False`
> TO：
> `for layer in model.layers[:168]: model.trainable = False for layer in model.layers[168:]: model.trainable = True`
> It works!
> I don't think there is a problem with original codes, but I haven't figured out what's the problem yet.

Hi Thank you for your solution!
I just check your code and seems that your code will make all model parameters trainable?
since you use `model.trainable = True` but not `layer.trainable = True`.

**Comment by devinhee** [comment]
> > > Hi Thank you very much for your pre-trained model!
> > > I am gathering some new data and using your code to fine-tune.After 100 epochs, I get a new model (one weird thing is that the size of new model reduce a lot.....). However when I test it, it seems that new model behave exactly same as the pre-trained one(same acc,same recall and same precision for each class).
> > > Any idea why it happens? I am checking your code and seems that you only set one layer trainable?
> > 
> > 
> > I have same issue with your, and I suggest you read Keras official documents.
> > I replace this code:
> > `model.trainable = False set_trainable = False for layer in model.layers: if layer.name == 'conv2d_56': set_trainable = True if set_trainable: layer.trainable = True else: layer.trainable = False`
> > TO：
> > `for layer in model.layers[:168]: model.trainable = False for layer in model.layers[168:]: model.trainable = True`
> > It works!
> > I don't think there is a problem with original codes, but I haven't figured out what's the problem yet.
> 
> Hi Thank you for your solution!
> I just check your code and seems that your code will make all model parameters trainable?
> since you use `model.trainable = True` but not `layer.trainable = True`.

Yes, it should be `layer.trainable = True`, it works for me. And I also try `model.trainable = True`, it seems to work out too. 
Here is where I found the solution: (https://keras.io/zh/applications/) . Under the 'Fine-tune InceptionV3 on a new set of classes'

**Comment by PravinVedurla** [comment]
@darcula1993 Thanks for raising this issue. Is there any update regarding the benchmarks of the model you've trained? Have you compared it with the available ones?
